francis/__main__.py

```diff
@@ -238,14 +247,23 @@ def listen(audio_sample, model_path):
             exit(1)
 
     print("predicting ...")
-    predictions = around(the_model.predict(spectrograms))
+    try:
+        predictions = the_model.predict(spectrograms)
+    except:
+        if len(the_df) == 0:
+            print(
+                "Error, data frame empty: input audio was completely filtered out in preprocessing and/or the original audio is silent"
+            )
+        exit(1)
+
+    print(around(the_model.predict(spectrograms), decimals=3))
 
     print("load categories")
     categories = io.load_categories(model_path)
 
     print("predictions ..")
-    #  print(model_adaptor.adapt_predictions(predictions, categories))
-    print(model_adaptor.summarize_predictions(predictions, categories))
+    bird_prediction = model_adaptor.predicted_bird_timestamp(predictions, categories)
+    print(model_adaptor.prediction_string_format(bird_prediction, categories))
 
```

Comment:
thanks for splitting this up to, this makes it easier to write tests for the logic in future.